py/types.py

```diff
@@ -0,0 +1,16 @@
+from collections.abc import Sequence, Mapping
+from typing import TypedDict, Protocol
+
+
+class AIProvider(Protocol):
+
+  def __init__(self, config: Mapping[str, str|float|dict]) -> None:
+      pass
+
+  def request(self, messages: Sequence[Message]) -> List[str]:
```

Comment:
I'm not very familiar with Python's typing, so I'd like to ask: why use a Sequence instead of a List here?

Reply:
I'm not very familiar with them either. I did it based on the [preference](https://docs.python.org/3/library/typing.html#typing.List) of using `Sequence` or `Iterable` for arguments. That would allow for tuples and lists, maybe some other structures?